ethereum_history_api/circuits/lib/src/account_with_storage.nr

```diff
@@ -0,0 +1,73 @@
+use dep::proof::{MAX_ACCOUNT_STATE_LENGTH, KEY_LENGTH};
+
+use crate::header::get_header;
+use crate::misc::types::{Address, Bytes32};
+use crate::verifiers::account::verify_account;
+use crate::verifiers::storage::verify_storage_values;
+
+global STATE_PROOF_LEN = 4788; // = 9 (MAX_STATE_PROOF_LEVELS) * 532 (MAX_TRIE_NODE_LENGTH)
+global STORAGE_PROOF_LEN = 3724; // = 7 (MAX_STORAGE_PROOF_LEVELS) * 532 (MAX_TRIE_NODE_LENGTH)
+
+struct Account {
+    nonce: Field,
+    balance: Field,
+    storage_root: Bytes32,
+    code_hash: Bytes32,
+}
+
+struct StateProof {
+    key: Address,
+    value: [u8; MAX_ACCOUNT_STATE_LENGTH],
+    proof: [u8; STATE_PROOF_LEN],
+    depth: Field,
+}
+
+struct StorageProof {
+    key: Bytes32,
+    value: Bytes32,
+    proof: [u8; STORAGE_PROOF_LEN],
+    depth: Field
+}
+
+// For now oracle does not support returning array of arrays so at the moment we support only one storage proof.
+// When https://github.com/noir-lang/noir/issues/4498 is resolved we can change to `StateAndStorageProof<N>` and `storage_proof: [StorageProof; N]`.
+struct StateAndStorageProof {
+    account: Account,
+    state_proof: StateProof,
+    storage_proof: StorageProof
+}
+
+struct StorageWithinBlock<N> {
+    block_hash: Bytes32,
+    account: Account,
+    values: [Bytes32; N],
+}
+
+pub fn get_account_with_storage(
+    block_number: Field,
+    address: Address,
+    storage_key: Bytes32
+) -> StorageWithinBlock<1> {
+    let header = get_header(block_number);
```

Comment:
Same for header

Reply:
But in this case it will require the following code 
```rust
let BlockHeaderPartial { number: _number, hash, state_root, transactions_root: _transactions_root, receipts_root: _receipts_root } = get_header(block_number);
```
we need to map `number` to `_number` otherwise we will get `unused variable` warning. We can't skip this because we will get `error: missing field: number`
all in all IMO `let header = get_header(block_number);` is shorter and more concize.

Reply:
Agree. We can try also `number: _` or `let BlockHeaderPartial { hash, _} = ...`
Not sure if any of those work

Reply:
OK, works, changed to 
```rust
let BlockHeaderPartial { number: _, hash, state_root, transactions_root: _, receipts_root: _ } = get_header(block_number);

```